### PR TITLE
Add buffer pool stat to flex counter architecture

### PR DESCRIPTION
**Pull request by wendani**
1. Add control path to BUFFER_POOL_PLUGIN_LIST
2. Add metadata deserialization support for buffer pool stats
Deserialize buffer pool counter id list to SAI stats
3. Implement setBufferPoolCounterList
4. Implement removeBufferPool which is the removal counterpart to setBufferPoolCounterList
5. Add buffer pool lua script to plugins to run
6. Add buffer pool stat in data collection

### CODE REVIEW

**Comment by lguohan** [comment]
build failure, can you check?

File: syncd/syncd_flex_counter.cpp [hunk 1]
```diff
@@ -804,14 +924,16 @@ void FlexCounter::collectCounters(
...
-//        TODO: replace if with get_queue_stats_ext() call when it's fully supported
-//        Example:
-//        sai_status_t status = sai_metadata_sai_queue_api->get_queue_stats_ext(
-//                queueId,
-//                static_cast<uint32_t>(queueCounterIds.size()),
-//                queueCounterIds.data(),
-//                m_statsMode,
-//                queueStats.data());
+#if 0
+        TODO: replace if with get_queue_stats_ext() call when it is fully supported
```
Comment (mykolaf) [inline syncd/syncd_flex_counter.cpp]: Why use #if 0?
If you are going to use it, add '//' in L#928 and remove L#929 entirely

File: syncd/syncd_flex_counter.cpp [hunk 2]
```diff
@@ -58,6 +59,14 @@ FlexCounter::RifCounterIds::RifCounterIds(
     SWSS_LOG_ENTER();
 }
 
+FlexCounter::BufferPoolCounterIds::BufferPoolCounterIds(
+        _In_ sai_object_id_t bufferPool,
+        _In_ const std::vector<sai_buffer_pool_stat_t> &bufferPoolIds):
+    bufferPoolId(bufferPool), bufferPoolCounterIds(bufferPoolIds)
+{
+    SWSS_LOG_ENTER();
```
Comment (kcudnik) [inline syncd/syncd_flex_counter.cpp]: why this is needed if empty ?

**Comment by lguohan** [comment]
can you look at the build failure?

File: syncd/syncd_flex_counter.cpp (hunk 2 above)
Comment (wendani) [reply syncd/syncd_flex_counter.cpp]: This is constructor. We use data member initialization syntax

**Comment by lguohan** [comment]
retest this please

**Comment by lguohan** [comment]
retest this please

**Comment by lguohan** [comment]
retest this please

File: syncd/syncd_flex_counter.cpp (hunk 1 above)
Comment (wendani) [reply syncd/syncd_flex_counter.cpp]: It looks prettier in vim

**Comment by wendani** [comment]
retest this please

**Pull request merged**